Tasks/AzureResourceManagerTemplateDeploymentV3/operations/DeploymentScopeBase.ts

```diff
@@ -18,7 +19,27 @@ export class DeploymentScopeBase {
     }
 
     public async deploy(): Promise<void> {
-        await this.createTemplateDeployment();
+        try {
+            await this.createTemplateDeployment();
+        } catch (error) {
+            if((error as string).toLowerCase().indexOf("serviceprincipal") != -1) {
+                try {
+                    var spnName = await this.getServicePrincipalName();
+                    tl.warning(tl.loc("ServicePrincipalRoleAssignmentDetails", spnName, this.taskParameters.resourceGroupName));
+                } 
+                catch (err)
+                {
+                    tl.error(err);
```

Comment:
Add some prefix to error to indicate that the error occurred while getting SPN name. And log this message only for debug mode `tl.debug()`

Reply:
Done